walker.go

```diff
@@ -0,0 +1,84 @@
+package ftp
+
+import (
+	"fmt"
+	"strings"
+)
+
+//Walker traverses the directory tree of a remote FTP server
+type Walker struct {
+	serverConn *ServerConn
+	root       string
+	cur        item
+	stack      []item
+	descend    bool
+}
+
+type item struct {
+	path  string
+	entry Entry
+	err   error
+}
+
+// Next advances the Walker to the next file or directory,
+// which will then be available through the Path, Stat, and Err methods.
+// It returns false when the walk stops at the end of the tree.
+func (w *Walker) Next() bool {
+	if w.descend && w.cur.err == nil && w.cur.entry.Type == EntryTypeFolder {
+		list, err := w.serverConn.List(w.cur.path)
+		if err != nil {
+			w.cur.err = nil
```

Comment:
Why do you ignore the error ?

Reply:
That's a typo... should be 
`w.cur.err = err`